3D/polygon/polygon_from_sprite.py

```diff
@@ -0,0 +1,445 @@
+    points = pointsFromPng(adress,sx,sy,ceil)
+    
+    pxa = points[0]
+    pya = points[1]
+    
+    npict = len(pxa)
+    
+    ppxs    = []
+    ppys    = []
+    polyall = []
+    
+    # List of values for minlen and maxlen. Both list should have len of llist
+    minlist = [5,4,3,2,1]
+    maxlist = [10,8,6,4,1.5]
+    llist = 5
+    
+    # Adapt minlist and maxlist in order to match presripted values
+    j = 0
+    for i in range(llist):
+        if minlist[j] > minlen or maxlist[j] > maxlen:
+            del(minlist[j])
+            del(maxlist[j])
+            j -= 1
+        j += 1
+        if j >= len(minlist):
+            break
+    
+    for ppi in range(npict):
+        px = pxa[ppi]
+        py = pya[ppi]
+
+        for j in range(len(minlist)):
+            stop = 1
+            
+            pplg    = singlePolygonFromPng(px,py,minlist[j],maxlist[j],ppi)
+            polygon = pplg[0]
+            ppx     = pplg[1]
+            ppy     = pplg[2]
+            
+            # Some checks
+            if len(polygon)==1001:
+                print('refining sprite '+str(ppi))
+                stop = 0
+                
+            elif abs(max(ppx)-max(px)) > minlen or abs(min(ppx)-min(px)) > minlen \
+              or abs(max(ppy)-max(py)) > minlen or abs(min(ppy)-min(py)) > minlen:
+                print('Polygon is not precise enough. Refining sprite '+str(ppi))
+                stop = 0
+                
+            if stop:
+                break
+
+        polysim = simplifyPolygon(polygon,ppx,ppy,math.pi/16) # Simplify the polygon
+        
+        ppxs.append(polysim[1])
+        ppys.append(polysim[2])
+        polyall.append(polysim[0])
+
+    return (points,(polyall,ppxs,ppys))
+    
+# Generates a XML file that contains the polygon
+def generateXML(polygon,adress):
+    
+    poly = polygon[0]
+    px   = polygon[1]
+    py   = polygon[2]
+    
+    nb   = len(poly)
+    
+    polygons = ET.Element('polygons')
+    
+    for i in range(nb):
+        ppx   = px[i]
+        ppy   = py[i]
+        npoly = len(ppx)
+        
+        polyg = ET.SubElement(polygons,'polygon')
+        x = ET.SubElement(polyg,'x')
+        y = ET.SubElement(polyg,'y')
+        
+        strx = str(ppx[0])
+        stry = str(ppy[0])
+        for j in range(npoly-1):
+            strx = strx + ',' + str(ppx[j+1])
+            stry = stry + ',' + str(ppy[j+1])
+
+        x.text = strx #strx
+        y.text = stry
+        
+    mydata = ET.tostring(polygons, encoding="UTF-8", method="xml")
+    mydata = pretty.parseString(mydata)
+    mydata = mydata.toprettyxml(indent="\t",encoding="UTF-8")
+
+    myfile = open(adress, "w")
+    myfile.write(mydata.decode("utf-8") )
+    myfile.close()
+
+
+if __name__ == "__main__":
+    #pntNplg = polygonFromPng('../../../naev/dat/gfx/ship/admonisher/admonisher.png',8,8,150,3,6)
+    pntNplg = polygonFromPng('../../../naev/dat/gfx/outfit/space/caesar.png',6,6,150,2,4)
```

Comment:
Having to manually edit this each time a polygon is generated, and tune the parameters is counterproductive to the aim I stated in naev#11.

Could the script be modified to include the right parameters for each ship and be able to generate them all?